src/common/components/repository-input/index.js

```diff
@@ -33,6 +34,14 @@ export class RepositoryInput extends Component {
     return message;
   }
 
+  componentWillReceiveProps(nextProps) {
```

Comment:
I'll leave it as it is for now. I guess it's not worth putting to much work into that before we make all pieces work together as a flow.